packages/nextjs/pages/api/etl/index.ts

```diff
@@ -4,55 +4,61 @@ import type { NextApiRequest, NextApiResponse } from "next";
 import { OnchainMetricsByProject } from "~~/app/types/OSO";
 import dbConnect from "~~/services/mongodb/dbConnect";
 import ETLLog from "~~/services/mongodb/models/etlLog";
-import GlobalScore from "~~/services/mongodb/models/globalScore";
-import Metric, { MetricNames, Metrics } from "~~/services/mongodb/models/metric";
+import GlobalScore, { TempGlobalScore } from "~~/services/mongodb/models/globalScore";
+import Metric, { Metrics } from "~~/services/mongodb/models/metric";
 import Project from "~~/services/mongodb/models/project";
-import ProjectMetricSummary from "~~/services/mongodb/models/projectMetricSummary";
-import ProjectScore, { IProjectScore } from "~~/services/mongodb/models/projectScore";
+import ProjectMetricSummary, { TempProjectMetricSummary } from "~~/services/mongodb/models/projectMetricSummary";
+import ProjectScore, { IProjectScore, TempProjectScore } from "~~/services/mongodb/models/projectScore";
+
+export const maxDuration = 300;
+export const runtime = "nodejs";
 
 export default async function handler(req: NextApiRequest, res: NextApiResponse) {
   if (req.method !== "GET") {
     return res.status(405).json({ message: "Method not allowed." });
   }
-  const weightings = weightingsJSON as { [key in keyof Metrics]: number };
   try {
-    await dbConnect();
+    const mongooseConnection = await dbConnect();
     // Log the ETL process
-    const existingLog = await ETLLog.findOne({ status: "pending" });
+    const existingLog = await ETLLog.findOne({}, {}, { sort: { _id: -1 } });
     if (existingLog) {
-      return res.status(400).json({ message: `ETL Process already running as of ${existingLog.date}` });
+      if (existingLog.status === "pending") {
+        return res.status(400).json({ message: `ETL Process already running as of ${existingLog.date}` });
+      }
+      if (existingLog.status === "success" && isTooEarly(existingLog.date)) {
+        return res.status(400).json({ message: `Too soon to run ETL Process. Last run: ${existingLog.date}` });
+      }
     }
+
     const log = await ETLLog.create({ date: new Date(), status: "pending" });
     res.status(200).json({ result: "ETL Process Started" });
-    // Start a session
-    const session = await startSession();
 
-    // Start a transaction
-    // session.startTransaction();
+    // Get weights from JSON file
+    const weightings = weightingsJSON as { [key in keyof Metrics]: number };
+
     try {
-      // Clear out the previous data
-      console.log("Clearing previous data");
-      await GlobalScore.deleteMany({}, { session });
-      await ProjectScore.deleteMany({}, { session });
-      await ProjectMetricSummary.deleteMany({}, { session });
```

Comment:
Should we clear Temp models at the beginning of the process? In case there's an error mid-way through and it needs to be run again?

Reply:
Not a bad idea.